packages/lib-algorand/index.ts

```diff
@@ -0,0 +1,85 @@
+import { DfnsApiClient, DfnsError } from '@dfns/sdk'
+import { GetWalletResponse, GenerateSignatureResponse } from '@dfns/sdk/types/wallets'
+
+import { encodeObj, Transaction, EncodedSignedTransaction, decodeObj } from 'algosdk'
+export type DfnsWalletOptions = {
+  walletId: string
+  dfnsClient: DfnsApiClient
+}
+
+const hexToBuffer = (hex: string): Buffer => {
+  return Buffer.from(hex.replace(/^0x/, ''), 'hex')
+}
+
+type WalletMetadata = GetWalletResponse
+
+const assertSigned = (res: GenerateSignatureResponse) => {
+  if (res.status === 'Failed') {
+    throw new DfnsError(-1, 'signing failed', res)
+  } else if (res.status !== 'Signed') {
+    throw new DfnsError(
+      -1,
+      'cannot complete signing synchronously because this wallet action requires policy approval',
+      res
+    )
+  }
+}
+
+export class DfnsWallet {
+  private readonly dfnsClient: DfnsApiClient
+
+  private constructor(private metadata: WalletMetadata, options: DfnsWalletOptions) {
+    this.dfnsClient = options.dfnsClient
+  }
+
+  public static async init(options: DfnsWalletOptions) {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    if (res.network !== 'Algorand' && res.network !== 'AlgorandTestnet') {
+      throw new DfnsError(-1, 'wallet is not bound to Algorand or AlgorandTestnet', {
+        walletId,
+        network: res.network,
+      })
+    }
+
+    return new DfnsWallet(res, options)
+  }
+
+  public get address(): string {
+    return this.metadata.address!
+
+  }
+
+  public async signTransaction(transaction: EncodedSignedTransaction): Promise<EncodedSignedTransaction>;
+  public async signTransaction(transaction: Transaction): Promise<EncodedSignedTransaction>;
+
+  async signTransaction(transaction: EncodedSignedTransaction | Transaction): Promise<EncodedSignedTransaction> {
+    if (!('txn' in transaction)) {
```

Comment:
this is a better check since Transaction is a class

```suggestion
    if (transaction instanceof Transaction) {
```

Reply:
Didn't know this pattern, thanks!

EDIT: it doesn't work. Apparently, when using `instanceof` with class that inherits from another class (or a chain inheritance), the check might not behave as expected.